### ISSUE TITLE
"no description" when bark run

### ISSUE THREAD

**Issue by NickAnastasoff** [gitmylo/bark-voice-cloning-HuBERT-quantizer#1]
I have tried to create an npz, although I think I have done something wrong. I have gotten bark running up until generate_coarse:
`Exception has occurred: AssertionError
exception: no description
  File "/Users/nickanastasoff/Desktop/bark test/bark/bark/generation.py", line 573, in generate_coarse
    round(x_coarse_history.shape[-1] / len(x_semantic_history), 1)
  File "/Users/nickanastasoff/Desktop/bark test/bark/bark/api.py", line 54, in semantic_to_waveform
    coarse_tokens = generate_coarse(
  File "/Users/nickanastasoff/Desktop/bark test/bark/bark/api.py", line 113, in generate_audio
    out = semantic_to_waveform(`

[customHuburt.txt](https://github.com/gitmylo/bark-voice-cloning-HuBERT-quantizer/files/11523469/customHuburt.txt)
This is what I used to make the npz. Im pretty sure the issue is with `fine_prompts = codes`  but im not sure what else to do.

**Comment by gitmylo** [comment]
This doesn't seem to be an issue with my repository. This repository exclusively extracts semantics.

Also, i was not able to reproduce the issue, your code worked fine on my side.
* Do you have the right path for your models?
* Do you have the right version of encodec installed? (`pip install -f encodec`)
* Maybe your wav is invalid? (try using ffmpeg `ffmpeg -i 0520.wav audio.wav`)

**Comment by NickAnastasoff** [comment]
Thank you so much for your reply! sadly it still didn't work for me. How did you generate the npz? this is what I wrote, so its probably the issue: `
from encodec import EncodecModel
from encodec.utils import convert_audio

import torchaudio
import torch

# Instantiate a pretrained EnCodec model
model = EncodecModel.encodec_model_24khz()
# The number of codebooks used will be determined bythe bandwidth selected.
# E.g. for a bandwidth of 6kbps, `n_q = 8` codebooks are used.
# Supported bandwidths are 1.5kbps (n_q = 2), 3 kbps (n_q = 4), 6 kbps (n_q = 8) and 12 kbps (n_q =16) and 24kbps (n_q=32).
# For the 48 kHz model, only 3, 6, 12, and 24 kbps are supported. The number
# of codebooks for each is half that of the 24 kHz model as the frame rate is twice as much.
model.set_target_bandwidth(6.0)

# Load and pre-process the audio waveform
wav, sr = torchaudio.load("0520.wav")
wav = convert_audio(wav, sr, model.sample_rate, model.channels)
wav = wav.unsqueeze(0)

# Extract discrete codes from EnCodec
with torch.no_grad():
    encoded_frames = model.encode(wav)
codes = torch.cat([encoded[0] for encoded in encoded_frames], dim=-1)  # [B, n_q, T]

fine_prompt = codes

coarse = fine_prompt[:2, :]

import numpy

numpy.savez(semantic_prompt=semantic_tokens, fine_prompt=fine_prompt, coarse_prompt=coarse, file="pleasework.npz")`

**Comment by gitmylo** [comment]
You should probably wrap your code in code blocks (``` around your text) in the future.

I ran that code, and it created the file just fine. Can you send me the wav you're using? I think your input wav is a bit broken, and encodec can't load it.

Again, this issue is not really related to my repository here. But it's probably your wav file.

**Comment by NickAnastasoff** [comment]
Sorry for the wait!

I put my code into a jupyter notebook, and I still got the same problem! Ill link that, and my audio.wav is in it.

Thanks so much for your time!

[](https://colab.research.google.com/drive/1IA3c_R859nANerMARazCSrjc2UD3ws8A?usp=sharing)

**Comment by gitmylo** [comment]
You can't upload an audio file like that to google colab, since it's storage is not persistent.

Check if you can clone the file in [here](https://huggingface.co/spaces/GitMylo/bark-voice-cloning)

**Comment by gitmylo** [comment]
Oh, actually, i noticed this today.

`codes = torch.cat([encoded[0] for encoded in encoded_frames], dim=-1) # [B, n_q, T]`

should be

`codes = torch.cat([encoded[0] for encoded in encoded_frames], dim=-1).squeeze() # [B, n_q, T]`

**Comment by NickAnastasoff** [comment]
That makes sense!  The code I write is usually the problem 🤣 

Thanks so much!

**Comment by gitmylo** [comment]
> That makes sense! The code I write is usually the problem 🤣
> 
> Thanks so much!

That was actually something i was missing in the old version, plus the encodec example doesn't have it. So that's on me.

**Comment by NickAnastasoff** [comment]
`codes = torch.cat([encoded[0] for encoded in encoded_frames], dim=-1) # [B, n_q, T]`

should be

`codes = torch.cat([encoded[0] for encoded in encoded_frames], dim=-1).squeeze() # [B, n_q, T]`

And shorten audio file to under 10 seconds
